### PR TITLE
Add imperative editor script API

### LINKED ISSUE #8427: Add imperative editor script API
**Context:**
It's easier to mix and match shell invocations and in-editor transactions in a procedural, imperative style (as opposed to returning a list of actions). Related to #6810, depends on #8426.

**Task (REQUIRED):**
Add more editor script functions:
- `editor.transact()` - apply a collection of `set` actions at once, refresh evaluation context, continue execution
- `os.execute()` - execute a shell command, do resource sync, continue execution
- `editor.save()` - save unsaved changes, which might be necessary in some rare cases before calling `os.execute()`

**Expected outcome (REQUIRED):**
Our editor actions can be executed imperatively.

### PR DESCRIPTION
**Pull request by vlaaad**
This changeset adds imperative editor script APIs:
1. `editor.transact(txs)`: similar to [`"set"` action](https://defold.com/manuals/editor-scripts/#undoable-actions) — modify the editor's in-memory state.
2. `editor.execute(cmd...)`: similar to [`"shell"` action](https://defold.com/manuals/editor-scripts/#non-undoable-actions) — execute a shell script.
3. `editor.save()`: persist all unsaved changes to disk.

Technical notes:
I'm currently working on updating the editor script manual.

Fixes #8427

### CODE REVIEW

**Review by matgis — APPROVED** [review]
Looks good!

I wonder if we should ask the user to allow the editor to execute a command like we do with the `"shell"` action? Or perhaps stop doing that for the `"shell"` action?

Seems like we should have the same behavior for both, but I also kind of think asking to execute a command won't really protect users very much since a bad actor could just had their nefarious code in an innocuous-looking shell script that is part of the extension. 🤔

**Comment by AGulev** [comment]
Do we want or need documentation updates?

**Comment by vlaaad** [comment]
@AGulev see my message at the beginning of the PR:
> I'm currently working on updating the editor script manual.

File: editor/src/clj/editor/lua.clj [hunk 1]
```diff
@@ -360,11 +360,88 @@
...
+                                                 </li>
+                                                 <li>
+                                                   <code>\"discard\"</code>: the error output is discarded completely.
+                                                 </li>
+                                               </ul>
+                                             </li>
+                                           </ul>"}]
+                      :returnvalues [{:name "result"
+                                      :types ["nil" "string"]
+                                      :doc "If <code>out</code> option is set to <code>\"capture\"</code>, returns the output as string with trimmed trailing newlines. Otherwise, returns <code>nil</code>."}]
```
[the remark this answers is not in the document]
Comment (vlaaad) [reply editor/src/clj/editor/lua.clj:410]: Yeah, I think we will be able to do it later if needed!

**Pull request merged**